File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_map = {}
        for i in range(len(nums)):
            if nums[i] in hash_map:
                return [hash_map[nums[i]], i]
            else:
                num_to_find = target - nums[i]
                hash_map[num_to_find] = i

File: 0001-two-sum/0001-two-sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        
        let mut hash_map = HashMap::with_capacity(nums.len());
        
        for (i, &num) in nums.iter().enumerate() {
            match hash_map.get(&num) {
                Some(j) => return vec![i as i32, *j as i32],
                None => {
                    hash_map.insert(target - num, i);
                }
            }
        }
        
        unreachable!()
    }
}